src/main/java/org/mskcc/oncokb/curation/service/FdaSubmissionQueryService.java

```diff
@@ -0,0 +1,163 @@
+        StringFilter stringFilter = new StringFilter();
+        stringFilter.setContains(query);
+        criteria.setDeviceName(stringFilter);
+        criteria.setNumber(stringFilter);
+        criteria.setSupplementNumber(stringFilter);
+        return findByCriteria(criteria, page);
+    }
+
+    /**
+     * Return a {@link List} of {@link FdaSubmission} which matches the criteria from the database.
+     * @param criteria The object which holds all the filters, which the entities should match.
+     * @return the matching entities.
+     */
+    @Transactional(readOnly = true)
+    public List<FdaSubmission> findByCriteria(FdaSubmissionCriteria criteria) {
+        log.debug("find by criteria : {}", criteria);
+        final Specification<FdaSubmission> specification = createSpecification(criteria);
+        return fdaSubmissionRepository.findAll(specification);
+    }
+
+    /**
+     * Return a {@link Page} of {@link FdaSubmission} which matches the criteria from the database.
+     * @param criteria The object which holds all the filters, which the entities should match.
+     * @param page The page, which should be returned.
+     * @return the matching entities.
+     */
+    @Transactional(readOnly = true)
+    public Page<FdaSubmission> findByCriteria(FdaSubmissionCriteria criteria, Pageable page) {
+        log.debug("find by criteria : {}, page: {}", criteria, page);
+        final Specification<FdaSubmission> specification = createSpecification(criteria);
+        return fdaSubmissionRepository.findAll(specification, page);
+    }
+
+    /**
+     * Return the number of matching entities in the database.
+     * @param criteria The object which holds all the filters, which the entities should match.
+     * @return the number of matching entities.
+     */
+    @Transactional(readOnly = true)
+    public long countByCriteria(FdaSubmissionCriteria criteria) {
+        log.debug("count by criteria : {}", criteria);
+        final Specification<FdaSubmission> specification = createSpecification(criteria);
+        return fdaSubmissionRepository.count(specification);
+    }
+
+    /**
+     * Function to convert {@link FdaSubmissionCriteria} to a {@link Specification}
+     * @param criteria The object which holds all the filters, which the entities should match.
+     * @return the matching {@link Specification} of the entity.
+     */
+    protected Specification<FdaSubmission> createSpecification(FdaSubmissionCriteria criteria) {
+        Specification<FdaSubmission> specification = Specification.where(null);
+        if (criteria != null) {
+            // This has to be called first, because the distinct method returns null
+            if (criteria.getDistinct() != null) {
+                specification = specification.or(distinct(criteria.getDistinct()));
+            }
+            if (criteria.getId() != null) {
+                specification = specification.or(buildRangeSpecification(criteria.getId(), FdaSubmission_.id));
```

Comment:
What is `FdaSubmission_`?

Reply:
Reading through it, is metamodal generated in the runtime?

Reply:
Its during build time and is located under the `target` folder